shared/chat/conversation/input-area/normal/container.js

```diff
@@ -162,24 +170,25 @@ export default compose(
       }
     },
     componentWillReceiveProps(nextProps) {
-      // Fill in the input with an edit or unsent text
+      // Fill in the input with an edit, quote, or unsent text
       if (this.props._editingMessage !== nextProps._editingMessage) {
         const text =
           nextProps._editingMessage && nextProps._editingMessage.type === 'text'
             ? nextProps._editingMessage.text.stringValue()
             : ''
         this.props.setText('') // blow away any unset stuff if we go into an edit, else you edit / cancel / switch tabs and come back and you see the unsent value
         this.props.setText(text, true)
-
-        if (!isMobile) {
-          const i = this.props.inputGetRef()
-          if (i) {
-            setImmediate(() => {
-              i.moveCursorToEnd()
-            })
-          }
+        !isMobile && this.props.inputMoveToEnd()
+      } else if (nextProps.quotedMessage && nextProps.quotedMessage !== this.props.quotedMessage) {
```

Comment:
this feels a little funny to me. i think we want a generic prop for data we want to inject into the input box. and if you quote something it should just set that field. so like 'injectedInput' or something. Then when you edit it should just use the value of the original message and set that part of the store also. we still need _editingMessage so we can highlight it and etc.